contracts/PetPark.sol

```diff
@@ -3,5 +3,58 @@ pragma solidity ^0.8.0;
 
 
 contract PetPark {
+    enum AnimalType{None,Fish,Cat,Dog,Rabbit,Parrot}
+    struct PetOwnerDetails{
+        AnimalType pet;
+        uint age;
+        Gender sex;
+    }
+    mapping(address => PetOwnerDetails)petOwner;
+    mapping(AnimalType => uint)public animalCounts;
+
+    enum Gender{Male,Female}
+    address owner;
+    event Added(AnimalType addedAnimal,uint count);
+    event Borrowed(AnimalType borrowedAnimal);
+    constructor(){
+        owner = msg.sender;
+    }
+    function add(AnimalType _animal, uint _count)public {
```

Comment:
This can be marked external since it is not being called anywhere internally

Reply:
I read about it correct me if I am wrong:
1. external uses less gas compare to public because public copies arguments in memory 
2. public should be called when the function needs to be called externally and internally in the contract but external only if it has to be accessed by the outside the contract only.
Am I missing something?

Reply:
You are right. You will also learn more about how gas is optimized using external later in the course